homeassistant/components/advantage_air/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""Advantage Air climate integration."""
+
+import collections.abc
+from datetime import timedelta
+import logging
+
+from advantage_air import advantage_air
+
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PORT
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import ADVANTAGE_AIR_RETRY, DOMAIN
+
+ADVANTAGE_AIR_SYNC_INTERVAL = 15
+ADVANTAGE_AIR_PLATFORMS = ["climate"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def update(original, updates):
+    """Deep update a dictionary."""
+    for key, val in updates.items():
+        if isinstance(val, collections.abc.Mapping):
+            original[key] = update(original.get(key, {}), val)
+        else:
```

Comment:
I've learned that else is 'less desired', so you could rewrite to:
```
for key, val ...
   original[key] = val
   if isinstance(...):
        original[key] = ...
```

Reply:
Cant do that, it completely breaks how the function works.

original = {"a": 1, "b": {"c": 1, "d": {"e": 1}}}
updates = {"b": {"z": 1}}

current function {'a': 1, 'b': {'c': 1, 'd': {'e': 1}, 'z': 1}}
no else function {'a': 1, 'b': {'z': 1}}

Reply:
I read it as:

```
if True:
  a = b
else:
  a =c
```

which should be the same as stating it 'unless' 

```
a = c 
if True:
  a = b
```

Personally I don't mind if's but mileage may vary, lets leave it as it is for now